### PR TITLE
README updates

### PR DESCRIPTION
**Pull request by benesjan**
# Description

@critesjosh mentioned that people complained about lacking description of bridgeId on Twitter. This PR hopefully fixes it. I've noticed that basically the whole readme needed an update so I rewrote it all. I sneaked in a fix of linting errors of the IDefiBridge interface because it was referenced in the readme and it was frustrating to not have it updated.

While updating the README I've stumbled upon a few suspicious things:

1. The table containing info about which bits in the bridgeId bit-string correspond to which data didn't corresponds to how it is actually done in the [encodeBridgeId function](https://github.com/AztecProtocol/aztec-connect-bridges/blob/5c32f02782d5649287c626fdd01afb7591068578/src/test/aztec/base/BridgeTestBase.sol#L197). I updated the table based on that.
2. It seems that the bitConfig as encoded in encodeBridgeId function doesn't contain all the information which was in the table. @LHerskind why do you encode only info about whether inputAssetB and outputAssetB are used but not info about which assets are virtual?
3. [EncodeBridgeIdTest](https://github.com/AztecProtocol/aztec-connect-bridges/blob/master/src/test/aztec/base/EncodeBridgeId.t.sol) is empty. @LHerskind do you plan on finishing it or should we remove it?
4. If the encoding in the table as I updated it is correct why are there 8 leftmost bits unused? Why don't we instead use them for auxData making auxData uint72?
5. If the bitConfig is supposed to be encoded as it is in the bitConfig table in the readme why do we make it 32 bits long? Based on the table 6 bits should be sufficient.

In my opinion using the name `bridgeId` in this context is just bad. When I hear `bridgeId` I imagine an actual id of the bridge contract (called `bridgeAddressId` here). I propose renaming it to `encodedInteraction` or `interactionInfo` (or something similar) once convenient.

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [x] There are no unexpected formatting changes, or superfluous debug logs.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewers next convenience.

### CODE REVIEW

**Comment by critesjosh** [comment]
+1 jan, i think renaming bridge id to `encodedInteraction` makes a lot of sense, especially since `BridgeId` means something completely different in the SDK.

**Comment by critesjosh** [comment]
@benesjan i added some more description to the deployed bridges table and included a link to the relevant docs page

**Comment by LHerskind** [comment]
1. I think the readme is just outdated, as you will see from below, we already have this data from the inputAssets etc. Alave an openingnonce and so, the old bridge id structure, seemed to have an opening nonce and only one input asset. The `RollupProcessor::getBridgeData()` is the best place to use as base. 
2. The `bitConfig` only really use 2 bits (currently), `secondInputInUse` and `secondOutputInUse`, whether the asset is virtual or not you already have from the asset ids. Take a look at `getBridgeData`. The `BridgeData` that we can deduce from it however, stores the `firstInputVirtual` as a separate flag for easy access etc. 
   https://github.com/AztecProtocol/aztec-connect/blob/b2103376608e46ffe50cf56f9ca5ce031f34c671/blockchain/contracts/RollupProcessor.sol#L1634
3. Think I just forgot it. It is implicitly checked through the use as wrong encoding would make other tests break, but probably fine to update it anyway. 
4. We are using 248 bits because the circuits don't support values of a full 256 bits, so 248 is the largest multiple of 8 (whole bytes) that we can use as input. 
5. The bitconfig is large for future proofing. It gives us a buffer If we need to use more bits for the input values down the line, e.g., say another bitflag for NFT's etc.
I think the naming would make a lot of sense to change. And agree that `encodedInteraction` would be a fine suggestion, much clearer and easier to follow, especially as there atm are two bridgeIds where one is a subset of the other :sweat:

**Review by LHerskind — APPROVED** [review]
Very minor comment

File: README.md [hunk 1]
```diff
@@ -5,86 +5,130 @@
 
 ## How to contribute
 
-This repo has been built with Foundry. Given the inter-connected nature of Aztec Connect Bridges with existing mainnet protocols, we decided Foundry / forge offered the best support for testing. This repo should make debugging, mainnet-forking, impersonation and gas profiling simple. It makes sense to test Solidity contracts with Solidty not with the added complication of Ethers / Typescript.
+This repo has been built with Foundry.
+Given the interconnected nature of Aztec Connect Bridges with existing mainnet protocols, we decided Foundry / forge offered the best support for testing.
```
[the remark this answers is not in the document]
Comment (benesjan) [reply README.md]: Addressed in a4ec3f2b4f4ee9d9a59e6ad190d9ef05223c875d

File: README.md [hunk 2]
```diff
@@ -5,86 +5,130 @@
...
+This repo should make debugging, mainnet-forking, impersonation and gas profiling simple.
+It makes sense to test Solidity contracts with Solidity, not with the added complication of Ethers / Typescript.
 
 ## Writing a bridge
 
-Developing a bridge is simple and permissionless. It is done entirely in Solidity and without any knowledge of the underlying cryptography Aztec uses. Users of your bridge will get the full benefits of ironclad privacy and 10-30x gas savings. Simply follow the steps below to get started.
+Developing a bridge is simple and permissionless.
```
[the remark this answers is not in the document]
Comment (benesjan) [reply README.md]: Addressed in a4ec3f2b4f4ee9d9a59e6ad190d9ef05223c875d

**Pull request merged**